Xamarin.Forms.Core/Internals/ResourceLoader.cs

```diff
@@ -5,8 +5,17 @@ namespace Xamarin.Forms.Internals
 {
 	public static class ResourceLoader
 	{
+		static Func<AssemblyName, string, string> resourceProvider;
+
 		//takes a resource path, returns string content
-		public static Func<AssemblyName, string, string> ResourceProvider { get; internal set; }
+		public static Func<AssemblyName, string, string> ResourceProvider {
+			get => resourceProvider;
+			internal set {
+				DesignMode.IsDesignModeEnabled = true;
```

Comment:
It's somewhat unfortunate that there is a non-evident side-effect to setting a `ResourceLoader.ResourceProvider` (a public API) that cannot be reversed via another public API (since `IsDesignModeEnabled` has an internal setter).

Coupling the two is not necessarily intuitive and the asymmetry could cause problems (i.e. for the live reload, we do set the `ResourceProvider` but will not want to be considered `DesignModeEnabled` since the actual app is running.

I'd suggest we go with the [principle of least surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) and fully decouple one from the other, making the setting of `DesignModeEnabled` explicit and publicly settable.

Reply:
Unfortunately the discussion about this was lost when the PR was re-created.

The tl;dr is that public settable doesn't make sense because then anyone can (and will) (re-)set it.

We, as toolkit owners, can use reflection to set it and/or use any other mechanism which allows it to be set and never unset.

The purpose of setting it automatically inside the APIs created specifically for the previewer/designer is that we then this boolean becomes usable/accurate for every existing release of VS and VSM. The alternative is that we will have a bool that will be wrong for the vast majority of people for the next year.

Reply:
I'm not sure how taking advantage of this new boolean can happen for existing releases of VSM unless you ship an updated one... In which case, you can just have the updated VSM set both at the same time....

Reply:
The boolean will be correctly set without requiring an IDE update if we continue with the mechanism proposed in this PR.